### ISSUE TITLE
Backdoor appears broken with new dependencies and/or Python 3

### ISSUE THREAD

**Issue by stefano-maggiolo** [cms-dev/cms#834]
Looks like some change in the parameters of gevent's BaseServer.

```
Traceback (most recent call last):
  File "/home/stefano/github/cms/.venv/bin/cmsContestWebServer", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/stefano/github/cms/scripts/cmsContestWebServer", line 58, in <module>
    sys.exit(main())
  File "/home/stefano/github/cms/scripts/cmsContestWebServer", line 51, in main
    ask_contest=ask_for_contest).run()
  File "/home/stefano/github/cms/cms/io/web_service.py", line 108, in run
    Service.run(self)
  File "/home/stefano/github/cms/cms/io/service.py", line 331, in run
    self.start_backdoor()
  File "/home/stefano/github/cms/cms/io/service.py", line 282, in start_backdoor
    self.backdoor = BackdoorServer(backdoor_sock, locals={'service': self})
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/backdoor.py", line 106, in __init__
    StreamServer.__init__(self, listener, spawn=group, **server_args)
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/server.py", line 90, in __init__
    BaseServer.__init__(self, listener, handle=handle, spawn=spawn)
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/baseserver.py", line 105, in __init__
    self.set_listener(listener)
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/server.py", line 117, in set_listener
    BaseServer.set_listener(self, listener)
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/baseserver.py", line 124, in set_listener
    self.family, self.address = parse_address(listener)
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/baseserver.py", line 400, in parse_address
    return _parse_address(address)
  File "/home/stefano/github/cms/.venv/lib/python3.5/site-packages/gevent/baseserver.py", line 389, in _parse_address
    raise TypeError('Expected tuple or string, got %s' % type(address))
TypeError: Expected tuple or string, got <class '_socket.socket'>
```

**Comment by lerks** [comment]
First findings: the issue is that under py2 `_socket.socket(socket.AF_UNIX,
socket.SOCK_STREAM)` provides an `accept` method whereas it doesn't under
`py3.

**Comment by lerks** [comment]
Fixed in c0cf47a4b21b36a9ea42daf8b6e65fe69dc47c5d.
